gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -86,8 +86,12 @@ private boolean includeField(Field f, boolean serialize) {
   private List<String> getFieldNames(Field f) {
     SerializedName annotation = f.getAnnotation(SerializedName.class);
     if (annotation == null) {
-      String name = fieldNamingPolicy.translateName(f);
-      return Collections.singletonList(name);
+      String fieldName = fieldNamingPolicy.translateName(f);
```

Comment:
I wonder if the logic for the two cases (with and without the annotation) could share more code. Would it be possible to have a `mainName` and `alternates`, which are set to `annotation.value()` and `Arrays.asList(annotation.alternate())` when `@SerializedName` is present; or `fieldNamingPolicy.translateName(f)` and `fieldNamingPolicy.translateToAlternateNames(f)` when it isn't? Then the code on lines 99–106 could be shared, with the necessary changes for `alternates` to be a list rather than an array.

Reply:
I have merged the similar code together and simplified the method.